api/src/resolver_repo.ts

```diff
@@ -76,11 +76,12 @@ export async function myCollabRepos(_, __, {userId}) {
 }
 
 export async function repo(_, { id }, { userId }) {
+  // a user can only access a private repo if he is the owner or a collaborator
   const repo = await prisma.repo.findFirst({
     where: { OR: [
       { id, public: true },
-      { id, owner: { id: userId!} },
-      { id, collaboratorIds: { has: userId!} },
+      { id, owner: { id: userId || "undefined"} },
+      { id, collaboratorIds: { has: userId || "undefined"} },
```

Comment:
What is the purpose of the string "undefined" here?

Reply:
When a guest visits a public repo, the `userId` could be null, but we should still return the repo. Though `OR` relationship may be short-circuited by the condition `public==true`, a null `userId` still throw a type error for the following two conditions.